File: 04_isprime/IsPrime.py
import math

class IsPrime:
    def __init__(self, n):
        self.n = n
        self.x = int(math.sqrt(n))


    def findisprime(self):
        if self.n <=1:
            return False
        
        for i in range(2, self.x+1):
            if self.n%i == 0:
                return False
            
        return True
        


if __name__ == "__main__":
    result = IsPrime(100)
    output = result.findisprime()
    print(output)

File: 04_isprime/IsPrime.java


class IsPrime {
    private int x;
    private int n;

    public IsPrime(int n) {
        this.n = n;
        this.x = (int) Math.sqrt(n);
    }

    public boolean findisprime() {
        if (n <=1) {
            return false;
        }


        for (int i = 2; i<= x; i++) {
            if (n % i == 0) {
                return false;
            }
        }
        return true;
    }


    public static void main(String[] args) {
        IsPrime ip = new IsPrime(5);
        boolean output = ip.findisprime();
        System.out.println(output);
    }
}